src/rust/ensogl/lib/text/src/buffer/view/selection.rs

```diff
@@ -112,69 +251,124 @@ impl Group {
         Group::default()
     }
 
-    /// Add a region to the selection. This method implements merging logic.
+    /// Convert selections to carets by snapping end edges to start ones.
+    pub fn snap_selections_to_start(&self) -> Group {
+        let sorted_selections = self.sorted_selections.iter().map(|t| t.snap_to_start()).collect();
+        Self {sorted_selections }
+    }
+
+    /// Convert selections to carets by snapping start edges to end ones.
+    pub fn snap_selections_to_end(&self) -> Group {
+        let sorted_selections = self.sorted_selections.iter().map(|t| t.snap_to_end()).collect();
+        Self {sorted_selections }
+    }
+
+    /// Reference to the newest created selection if any.
+    pub fn newest(&self) -> Option<&Selection> {
+        self.sorted_selections.iter().max_by(|x,y| x.id.cmp(&y.id))
+    }
+
+    /// Reference to the oldest created selection if any.
+    pub fn oldest(&self) -> Option<&Selection> {
+        self.sorted_selections.iter().min_by(|x,y| x.id.cmp(&y.id))
+    }
+
+    /// Mutable reference to the newest created selection if any.
+    pub fn newest_mut(&mut self) -> Option<&mut Selection> {
+        self.sorted_selections.iter_mut().max_by(|x,y| x.id.cmp(&y.id))
+    }
+
+    /// Mutable reference to the oldest created selection if any.
+    pub fn oldest_mut(&mut self) -> Option<&mut Selection> {
+        self.sorted_selections.iter_mut().min_by(|x,y| x.id.cmp(&y.id))
+    }
+
+    /// Merge new selection with the group. This method implements merging logic.
     ///
     /// Two non-caret regions merge if their interiors intersect. Merely touching at the edges does
     /// not cause a merge. A caret merges with a non-caret if it is in the interior or on either
     /// edge. Two carets merge if they are the same offset.
     ///
     /// Performance note: should be O(1) if the new region strictly comes after all the others in
     /// the selection, otherwise O(n).
-    pub fn add(&mut self, region:Selection) {
-        let mut ix = self.selection_on_the_left_to(region.min());
-        if ix == self.sorted_regions.len() {
-            self.sorted_regions.push(region);
+    pub fn merge(&mut self, region:Selection) {
```

Comment:
Depending on the text editor. IntellJ merges them. VSCode, Atom, Sublime do not merge them. With our display style (we have visual separator between such selections) I like not merging more. I would;d keep it this way also because its so rare use case that no one really will care.